core.d.ts

```diff
@@ -0,0 +1,145 @@
+interface Util {
+	isString(v?: any): v is string
+	isNumber(v: any): v is number
+	isBoolean(v: any): v is boolean
+	isObject(v: any): v is object
+	isArray(v: any): v is Array<any>
```

Comment:
Ah, I thought it was doing what I wanted before because it works fine in your first case, typing it `a` as `string[]` within the if statement, but after I changed it I actually found I had one `b` case that caused problems - changed it now :)